torchtitan/__init__.py

```diff
@@ -0,0 +1,4 @@
+from torch._utils import _get_available_device_type, _get_device_module
+
+DEVICE_TYPE = _get_available_device_type()
+DEVICE_MODULE = _get_device_module(DEVICE_TYPE)
```

Comment:
Can we keep these in utils as helper functions? 
Having in global scope may lead to them being executed when torchtitan is just imported. Just want to be cautious about potential side effects of these calls (such as creating a device context).

Reply:
Yes, the helper function is added as below
# util.py
from torch._utils import _get_available_device_type, _get_device_module
def get_device_info():
    device_type = _get_available_device_type()
    device_module = _get_device_module(device_type)
    return device_type, device_module